src/components/AppMenu/AppMenu.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import styles from '@/components/AppMenu/AppMenu.module.scss'
+import { IconType } from 'react-icons'
+import { Text } from '../Text/Text'
+
+interface AppMenuProps {
+  icon?: IconType
+  text?: string
+  badge?: React.ReactElement
```

Comment:
Badge should only allow simple strings. We don't support and need passing components to the badge.